catalog/spatial/wfs/spatial-wfs-converter/src/main/java/org/codice/ddf/spatial/ogc/wfs/catalog/converter/impl/AbstractFeatureConverter.java

```diff
@@ -222,7 +223,44 @@ protected Metacard createMetacardFromFeature(HierarchicalStreamReader hreader,
                     LOGGER.debug("Setting metacard basic attribute: {} = {}",
                             reader.getNodeName(),
                             value);
-
+                    AttributeType expectedType = mc.getAttributeType(reader.getNodeName());
+                    if (expectedType.getBinding() != null && expectedType.getBinding() != value.getClass()) {
+                        //Attempt to convert the value
+                        switch(expectedType.getAttributeFormat()) {
+                        case BOOLEAN:
+                            value = Boolean.parseBoolean((String) value);
+                            break;
+                        case DOUBLE:
+                            value = Double.parseDouble((String) value);
+                            break;
+                        case FLOAT:
+                            value = Float.parseFloat((String) value);
+                            break;
+                        case INTEGER:
+                            value = Integer.parseInt((String) value);
+                            break;
+                        case LONG:
+                            value = Long.parseLong((String) value);
+                            break;
+                        case SHORT:
+                            value = Short.parseShort((String) value);
+                            break;
+                        case BINARY:
+                            value = Byte.parseByte((String) value);
+                            break;
+                        case OBJECT:
+                            break;
+                        case DATE:
+                            //TODO: Use SimpleDateFormat?
+                            value = new Date((String) value);
```

Comment:
WHAT THE HELL IS THIS?!?!?

This ctor on `Date` has been deprecated since Java 1.1. I'm going to go out on a limb and guess that's before most people touching this code base other than myself have been programming. This is wrong and completely broken.